### PR TITLE
WIP: Add lambda

### PR DESCRIPTION
**Pull request by kmcgill88**
My first untested attempt to add Lambda. I was hoping you could get an early look and provide any feedback.

Thanks!

### CODE REVIEW

File: lib/lambda.dart [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  /// https://docs.aws.amazon.com/lambda/latest/dg/API_Invoke.html
+  /// **Optional additional headers**
+  /// 'X-Amz-Log-Type': 'Tail',
+  /// 'X-Amz-Client-Context': base64.encode(utf8.encode('{"dart":true}'))
+  ///
+  /// functionName: arn:aws:lambda:us-west-2:123456789012:function:my-function <--format not working
+  /// functionName: my-function:$LATEST, my-function:v1 <--format not working
+  /// functionName: my-function <--Works!
```
Comment (kmcgill88) [inline lib/lambda.dart]: Not sure why these other formats didn't work. 🤷‍♂ Maybe it breaks the signing?

File: lib/src/request.dart [hunk 2]
```diff
@@ -31,11 +31,12 @@ class AwsResponse {
       : _body = body;
 
   /// A very lousy way to validate some of the common status codes.
+  /// https://http.cat/
   void validateStatus() {
-    if (statusCode == 200) return;
+    if (statusCode >= 200 && statusCode <= 207) return;
```
Comment (kmcgill88) [inline lib/src/request.dart]: `LambdaInvocationType.Event` returns a `202`.

File: examples/lambda.example.dart [hunk 3]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2016, project contributors. All rights reserved.
```
Comment (isoos) [inline examples/lambda.example.dart]: nit: 2019

**Comment by isoos** [comment]
Looks good, please also update Changelog and Readme, and pubspec.yaml (to increment the version).

**Comment by kmcgill88** [comment]
Done, Thank you!

**Comment by isoos** [comment]
Thanks for the PR, I've published 0.2.0.

**Pull request merged**